paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -494,11 +494,13 @@ internal class PaymentSheetViewModel @Inject internal constructor(
             val paymentMethodOptionsParams =
                 (paymentSelection.paymentMethodOptionsParams as? PaymentMethodOptionsParams.Card)
                     ?: PaymentMethodOptionsParams.Card()
-            paymentSelection.copy(
+            val newSelection = paymentSelection.copy(
                 paymentMethodOptionsParams = paymentMethodOptionsParams.copy(
                     cvc = cvcControllerFlow.value.fieldValue.value
                 )
             )
+            updateSelection(newSelection)
+            newSelection
```

Comment:
Was the selection not being updated before despite it being returned? Where is the returned value used?

Reply:
The return value was being used directly in `confirmPaymentSelection` just below